### PR TITLE
[BUG] "Read more" section title should not be shown on screen when there are no related articles

### PR DESCRIPTION
**Pull request by nambatee**
https://phabricator.wikimedia.org/T172906

### CODE REVIEW

File: Wikipedia/Code/WebViewController.m [hunk 1]
```diff
@@ -303,7 +303,8 @@ - (void)handleLateJavascriptTransformScriptMessage:(NSString *)messageString {
         [self.webView wmf_setPageProtected:!self.article.editable];
     } else if ([messageString isEqualToString:@"addFooterContainer"]) {
         [self.webView wmf_addFooterContainer];
-    } else if ([messageString isEqualToString:@"addFooterReadMore"]) {
+        // ns of articles with Read More pages equals 0.
+    } else if ([messageString isEqualToString:@"addFooterReadMore"] && self.article.ns == 0) {
```
Comment (joewalsh) [inline Wikipedia/Code/WebViewController.m]: should also check for `!self.article.isMain` - could add a computed readonly property `hasReadMore` to `MWKArticle` to use here and in `WMFArticleViewController`

Comment (nambatee) [inline Wikipedia/Code/WebViewController.m]: sounds great, will add that!

File: Wikipedia/Code/MWKArticle.m [hunk 2]
```diff
@@ -491,4 +491,11 @@ - (NSString *)articleHTML {
     return htmlStr;
 }
 
+#pragma mark Read More
+
+- (BOOL)hasReadMore {
+    // ns of articles with Read More pages equals 0.
+    return self && !self.isMain && self.ns == 0;
```
Comment (joewalsh) [inline Wikipedia/Code/MWKArticle.m]: The check for `self` is unnecessary, but doesn't hurt

File: Wikipedia/Code/WMFArticleViewController.m [hunk 3]
```diff
@@ -379,7 +379,7 @@ - (BOOL)hasAboutThisArticle {
     return self.article && !self.article.isMain;
 }
 
-- (void)getShareText:(void (^)(NSString *text))completion {
```
Comment (joewalsh) [inline Wikipedia/Code/WMFArticleViewController.m]: there was a `^C` between `)` and `c`

**Pull request merged**